rust-runtime/aws-smithy-runtime/src/client/http/body/minimum_throughput/http_body_0_4_x.rs

```diff
@@ -46,44 +91,27 @@ where
         let mut sleep_fut = this
             .sleep_fut
             .take()
-            .unwrap_or_else(|| this.async_sleep.sleep(this.options.check_interval()));
+            .unwrap_or_else(|| this.async_sleep.sleep(*this.resolution));
         if let Poll::Ready(()) = pin!(&mut sleep_fut).poll(cx) {
             tracing::trace!("sleep future triggered—triggering a wakeup");
             // Whenever the sleep future expires, we replace it.
-            sleep_fut = this.async_sleep.sleep(this.options.check_interval());
+            sleep_fut = this.async_sleep.sleep(*this.resolution);
 
             // We also schedule a wake up for current task to ensure that
             // it gets polled at least one more time.
```

Comment:
So, I would have thought sleep_fut wasn't required anymore since we are using the ThroughputCheckFuture externally. But if I remove it, download_stalls and download_too_slow fail. Do you understand why? 

Maybe it's just a matter of settings? I wonder if we could refactor that so we can remove that complexity from here.

Reply:
This is the download body, which _does not_ use ThroughputCheckFuture since it's not the orchestrator that's waiting on it. Rather, the customer is waiting on it in their code directly since it is the response body.